NPE when logging warnings through logback

When Kamon's initialization needs to log a warning and the Logback instrumentation is present a NPE is thrown because the Logback instrumentation is trying to read Kamon's config before it gets fully initialized. Stack trace:

```
Exception in thread "main" java.lang.ExceptionInInitializerError
    at kamon.instrumentation.logback.GetPropertyMapMethodInterceptor$.aroundGetMDCPropertyMap(LogbackInstrumentation.scala:92)
    at ch.qos.logback.classic.util.LogbackMDCAdapter.getPropertyMap(LogbackMDCAdapter.java)
    at ch.qos.logback.classic.spi.LoggingEvent.getMDCPropertyMap(LoggingEvent.java:306)
    at ch.qos.logback.classic.spi.LoggingEvent.prepareForDeferredProcessing(LoggingEvent.java:209)
    at ch.qos.logback.core.OutputStreamAppender.subAppend(OutputStreamAppender.java:223)
    at ch.qos.logback.core.OutputStreamAppender.append(OutputStreamAppender.java:102)
    at ch.qos.logback.core.UnsynchronizedAppenderBase.doAppend(UnsynchronizedAppenderBase.java:84)
    at ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttachableImpl.java:51)
    at ch.qos.logback.classic.Logger.appendLoopOnAppenders(Logger.java:270)
    at ch.qos.logback.classic.Logger.callAppenders(Logger.java:257)
    at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:421)
    at ch.qos.logback.classic.Logger.filterAndLog_0_Or3Plus(Logger.java:383)
    at ch.qos.logback.classic.Logger.warn(Logger.java:692)
    at kamon.Configuration.loadInitialConfiguration(Configuration.scala:75)
    at kamon.Configuration.$init$(Configuration.scala:27)
    at kamon.Kamon$.<init>(Kamon.scala:19)
    at kamon.Kamon$.<clinit>(Kamon.scala)
    at io.otrl.fare_finder.infrastructure.MainApp$.run(MainApp.scala:86)
    at cats.effect.IOApp.$anonfun$main$3(IOApp.scala:67)
    at cats.effect.internals.IOAppPlatform$.mainFiber(IOAppPlatform.scala:37)
    at cats.effect.internals.IOAppPlatform$.main(IOAppPlatform.scala:24)
    at cats.effect.IOApp.main(IOApp.scala:67)
    at cats.effect.IOApp.main$(IOApp.scala:66)
    at io.otrl.fare_finder.infrastructure.MainApp$.main(MainApp.scala:74)
    at io.otrl.fare_finder.infrastructure.MainApp.main(MainApp.scala)
Caused by: java.lang.NullPointerException
    at kamon.instrumentation.logback.LogbackInstrumentation$.readSettings(LogbackInstrumentation.scala:65)
    at kamon.instrumentation.logback.LogbackInstrumentation$.<init>(LogbackInstrumentation.scala:50)
    at kamon.instrumentation.logback.LogbackInstrumentation$.<clinit>(LogbackInstrumentation.scala)
    ... 25 more
```

Reported by @IgorCandido on [Gitter](https://gitter.im/kamon-io/Kamon?at=5e3c021f6f9d3d349824748c)

--- a/core/kamon-core/src/main/scala/kamon/Configuration.scala
+++ b/core/kamon-core/src/main/scala/kamon/Configuration.scala
@@ -19,11 +19,13 @@ package kamon
 import com.typesafe.config.{Config, ConfigFactory}
 import org.slf4j.LoggerFactory
 
+import scala.util.control.NonFatal
+
 /**
   * Exposes APIs to access and modify Kamon's configuration and to get notified of reconfigure events.
   */
 trait Configuration {
-  private val logger = LoggerFactory.getLogger(classOf[Configuration])
+  private val _logger = LoggerFactory.getLogger(classOf[Configuration])
   private var _currentConfig: Config = loadInitialConfiguration()
   private var _onReconfigureHooks = Seq.empty[Configuration.OnReconfigureHook]
 
@@ -43,7 +45,7 @@ trait Configuration {
       try {
         hook.onReconfigure(newConfig)
       } catch {
-        case error: Throwable => logger.error("Exception occurred while trying to run a OnReconfigureHook", error)
+        case NonFatal(t) => _logger.error("Exception occurred while trying to run a OnReconfigureHook", t)
       }
     })
   }
@@ -68,19 +70,24 @@ trait Configuration {
 
 
   private def loadInitialConfiguration(): Config = {
+    import System.{err, lineSeparator}
+
     try {
       ConfigFactory.load(ClassLoading.classLoader())
     } catch {
-      case t: Throwable =>
-        logger.warn("Failed to load the default configuration, attempting to load the reference configuration", t)
+      case NonFatal(t) =>
+        err.println("Kamon couldn't load configuration settings from your *.conf files due to: " +
+          t.getMessage + " at " + t.getStackTrace().mkString("", lineSeparator, lineSeparator))
 
         try {
           val referenceConfig = ConfigFactory.defaultReference(ClassLoading.classLoader())
-          logger.warn("Initializing with the default reference configuration, none of the user settings might be in effect", t)
+          err.println("Initializing Kamon with the reference configuration, none of the user settings will be in effect")
           referenceConfig
         } catch {
-          case t: Throwable =>
-            logger.error("Failed to load the reference configuration, please check your reference.conf files for errors", t)
+          case NonFatal(t) =>
+            err.println("Kamon couldn't load the reference configuration settings from the reference.conf files due to: " +
+              t.getMessage + " at " + t.getStackTrace().mkString("", lineSeparator, lineSeparator))
+
             ConfigFactory.empty()
         }
     }

